uat/steps/commons.py

```diff
@@ -0,0 +1,66 @@
+import os
+import t_utils
+import shutil
+
+from behave import step
+from pathlib import Path
+from constants import (
+    DEFAULT_AWS_REGION, DEFAULT_S3_BUCKET_PREFIX
+)
+
+
+@step('we make directory {dir_name}')
+def have_new_work_dir(context, dir_name):
+    cwd = Path(context.cwd or os.getcwd())
+    new_dir = cwd.joinpath(dir_name)
+    if not new_dir.exists():
+        new_dir.mkdir(parents=True, exist_ok=True)
+    assert new_dir.exists(), f"Directory {dir_name} does not exist in {cwd}"
+    os.chdir(str(new_dir))
+    context.cwd = os.getcwd()
+
+
+@step('we change directory {dir_name}')
+def change_work_dir(context, dir_name):
+    cwd = Path(context.cwd or os.getcwd())
+    new_dir = cwd.joinpath(dir_name)
+    assert new_dir.exists(), f"Directory {dir_name} does not exist in {cwd}"
+    os.chdir(str(new_dir))
+    context.cwd = os.getcwd()
+
+
+@step('we get s3 bucket name')
+def get_s3_bucket(context):
+    aws_account = t_utils.get_acc_num(DEFAULT_AWS_REGION)
+    context.last_s3_bucket = f"{DEFAULT_S3_BUCKET_PREFIX}-{DEFAULT_AWS_REGION}-{aws_account}"
+
+
+@step('we download {file_type} file {file_name} from s3')
+def download_s3_file(context, file_type, file_name):
+    file = f"{file_name}.{file_type}"
+    unique_file = f"{file_name}-{t_utils.random_id()}.{file_type}"
+    aws_account = t_utils.get_acc_num(DEFAULT_AWS_REGION)
+    bucket = f"{DEFAULT_S3_BUCKET_PREFIX}-{DEFAULT_AWS_REGION}-{aws_account}"
+    cwd = context.cwd if "cwd" in context else os.getcwd()
+    unique_file_path = Path(cwd).joinpath(unique_file).resolve()
+
+    s3_client = context.s3_client if "s3_client" in context else t_utils.create_s3_client(DEFAULT_AWS_REGION)
+    s3_client.download_file(bucket, file, str(unique_file_path))
+    assert unique_file_path.exists(), f"Couldn't find the downloaded {file} at {unique_file_path}"
+    context.last_file = unique_file_path
+
+
+@step('we unpack that {file_type} file into {dir_name} directory')
```

Comment:
should we tie it last_file always?

Reply:
BDD tests are generally contextual, so it has to be last_file in this case. However, we can rephrase the statement to:

- `we unpack last {file_type} file into {dir_name} directory`

Reply:
This is moot, as this step has been removed.